### ISSUE TITLE
geometry: a Box's realized cells depend on JAX_ENABLE_X64 — float32 node coordinates violate the documented half-open [lo, hi) convention at node-aligned faces

### ISSUE THREAD

**Issue by bk-squared** [bk-squared/rfx#802]
Found while re-measuring #589 (PR #799): the f64 replicate run 369367257270 failed its pre-solve structural test because `attempt2` and `attempt2_wide` were no longer junction-byte-identical under `JAX_ENABLE_X64=1`. Chasing it: **every Box/Cylinder mask on the fixture changes with the x64 flag** (attempt2: 5/5 entities, 7279 eps_r cells and 311 pec_mask cells differ between x64=0 and x64=1 builds — whole node planes).

## Mechanism
`rfx/api/_compile.py::_grid_coords` builds node coordinates as `(jnp.arange(n) - pad) * dx` in JAX's default float dtype, and `rfx/geometry/csg.py::_axis_mask` then applies the documented half-open test `(coords >= lo) & (coords < hi)`. In float32 the product `i * dx` lands a few 1e-10 m *below* the exact node value, so at faces declared on exact node multiples (the normal way geometry is written) the comparison flips relative to the convention the `Box` docstring promises ("node j belongs iff lo <= j·dx < hi; the hi face contributes no cell"):

Minimal repro (one Box, `domain=(12.5, 3.4, 3.9) mm`, `dx=100 µm`, `cpml_layers=8`, `Box((0,0,2.5e-3), (12.5e-3, 3.4e-3, 2.8e-3))`, script in PR #799's scratch — will be committed as a regression test):

```
x64=False coords dtype=float32 cells=13230 x nodes 0..125 y nodes 0..34 z nodes 25..27
  x hi=1.25e-02: node 125 coord=0.01249999925494194  coord<hi:True   <- hi face node INCLUDED (convention: excluded)
  y hi=3.40e-03: node 34  coord=0.003399999812245369 coord<hi:True   <- included
  z lo=2.50e-03: node 25  coord=0.0024999999441206455 coord>=lo:False <- lo face node EXCLUDED (convention: included)
  z hi=2.80e-03: node 28  coord=0.00279999990016222  coord<hi:True   <- included
x64=True  coords dtype=float64 cells=12750 x nodes 0..124 y nodes 0..33 z nodes 25..27
  every face exactly per the documented convention
```

So under the default precision (x64=0 — every f32 run in the repo) a node-aligned face is realized one node too far or too short depending on the float32 rounding of that particular `i*dx`, per face, per axis. Under x64=1 the convention holds exactly. This is not a tolerance question: two builds of the same declaration disagree by whole node planes, and which way they disagree is a rounding accident.

## Why it matters
- **Declared-vs-realized (#722, #729, #752 class)**: the fidelity reports in the #589 run show exactly this pattern — ground plane `x declared [0, 12500] -> realized [0, 12600]`, `y [0, 3400] -> [0, 3500]` (the hi-face node gained), substrate `z [2550, 2850] -> [2600, 2900]`, trace `y [1400, 2000] -> [1400, 2100]` (600 -> 700 µm). Part of what #722 catalogued as "half-open rounds up" is float32 coordinate error, not the convention. `fidelity_report` correctly *reports* the realized cells; the root cause is upstream of it.
- **Precision comparisons are confounded**: any f32-vs-f64 comparison (the #589 f64 replicate; cv09 pins x64=1 while cv11 pins x64=0) compares two different geometries. The #589 f64 replicate is BLOCKED on this.
- **Byte-identity chains and golden fixtures** were pinned on f32 realizations that violate the stated convention.

## Proposed fix (needs design + adversarial review; blast radius is large)
Compute node coordinates exactly: integer node indices compared against `round(face/dx)` (uniform grid), or float64 numpy coordinates independent of `jax_enable_x64`, keeping the traceable path only where coordinates are genuinely traced (differentiable mesh). Add a regression test: mask of a node-aligned Box is identical for x64=0 and x64=1 and matches the documented convention (fail-before-fix on the repro above). Then re-capture every realized-geometry-dependent snapshot/golden **as a separate, enumerated step** — nothing re-pinned silently — and re-read #722's per-script findings against the corrected realization.

Related: #722, #729, #752, #589 (blocked f64 replicate), PR #799 (where the structural test caught it).

**Comment by bk-squared** [comment]
Design → blast-radius probe → adversarial review done (verdict on the design: APPROVE_WITH_CHANGES). Evidence-only prototype and the full decision brief are in draft PR #803: one Box realizes 12750 cells under both x64 flags on both lanes; attempt2 arrays 0 cells differ across x64; new test file 10 failed on main → 15 passed on the prototype. Two things the issue text got wrong, now corrected there: (1) float64 alone is NOT exact — f64 k·dx lands one ulp below the typed decimal on 61–157/398 node-aligned faces at WR-90/patch cell sizes, so the fix needs an explicit node-snap tolerance; (2) the NU lane has its own copy of the defect (float32-cast cumsum nodes), so uniform and NU realizations currently disagree on the same declaration. Three new conventions (NODE_SNAP_REL = 1e-6·d_local, thin-sheet tie → lower node, Cylinder/Sphere closed-with-snap) need PI ratification; the enumerated re-pin list (7 hygiene pins, 2 snapshot variants, msl_s_matrix_golden.npy fixture, x64=1 cylinder references) is in the PR. Nothing re-captured, nothing loosened.
